src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -0,0 +1,233 @@
+        [Fact]
+        public void ArrayBasedConstructor_Int16() => TestArrayBasedConstructor<Int16>();
+        [Fact]
+        public void ArrayBasedConstructor_UInt32() => TestArrayBasedConstructor<UInt32>();
+        [Fact]
+        public void ArrayBasedConstructor_Int32() => TestArrayBasedConstructor<Int32>();
+        [Fact]
+        public void ArrayBasedConstructor_UInt64() => TestArrayBasedConstructor<UInt64>();
+        [Fact]
+        public void ArrayBasedConstructor_Int64() => TestArrayBasedConstructor<Int64>();
+        [Fact]
+        public void ArrayBasedConstructor_Single() => TestArrayBasedConstructor<Single>();
+        [Fact]
+        public void ArrayBasedConstructor_Double() => TestArrayBasedConstructor<Double>();
+
+        private void TestArrayBasedConstructor<T>() where T : struct
+        {
+            T[] values = GenerateRandomValuesForVector<T>(Vector<T>.Count).ToArray();
+            Vector<T> vector = new Vector<T>(values);
+            ValidateVector(vector,
+                (index, val) =>
+                {
+                    Assert.Equal(values[index], val);
+                });
+        }
+
+        [Fact]
+        public void ArrayIndexBasedConstructor_Byte() => TestArrayIndexBasedConstructor<Byte>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_SByte() => TestArrayIndexBasedConstructor<SByte>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_UInt16() => TestArrayIndexBasedConstructor<UInt16>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_Int16() => TestArrayIndexBasedConstructor<Int16>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_UInt32() => TestArrayIndexBasedConstructor<UInt32>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_Int32() => TestArrayIndexBasedConstructor<Int32>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_UInt64() => TestArrayIndexBasedConstructor<UInt64>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_Int64() => TestArrayIndexBasedConstructor<Int64>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_Single() => TestArrayIndexBasedConstructor<Single>();
+        [Fact]
+        public void ArrayIndexBasedConstructor_Double() => TestArrayIndexBasedConstructor<Double>();
+
+        private void TestArrayIndexBasedConstructor<T>() where T : struct
+        {
+            T[] values = GenerateRandomValuesForVector<T>(Vector<T>.Count * 2).ToArray();
+            int offset = Vector<T>.Count - 1;
+            Vector<T> vector = new Vector<T>(values, offset);
+            ValidateVector(vector,
+                (index, val) =>
+                {
+                    Assert.Equal(values[offset + index], val);
+                });
+        }
+
+        [Fact]
+        public void ArrayBasedConstructorWithLessElements_Byte() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayBasedConstructorWithLessElements<Byte>());
```

Comment:
`Assert.Throws<IndexOutOfRangeException>` could be moved inside of `TestArrayBasedConstructorWithLessElements` to avoid it being duplicated 10 times here.